src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -1,110 +1,431 @@
+
+        self.pubsub = None
 
     def reinit_data(self):
         """
         Re-initialize all data.
         """
-        device_metadata = self.statedb.get_all(self.statedb.STATE_DB, self.DEVICE_METADATA)
 
-        # TODO: Add support for unicode in swsssdk
-        chassis_serial_number = ""
+        # reinit cache
+        self.physical_classes_map = {}
+        self.physical_description_map = {}
+        self.physical_hw_version_map = {}
+        self.physical_serial_number_map = {}
+        self.physical_mfg_name_map = {}
+        self.physical_model_name_map = {}
+
+        # update interface maps
+        _, self.if_alias_map, _, _, _ = \
+            mibs.init_sync_d_interface_tables(SonicV2Connector())
+
+        device_metadata = mibs.get_device_metadata(self.statedb)
+        chassis_sub_id = (self.CHASSIS_ID, )
+        self.physical_entities = [chassis_sub_id]
+
         if not device_metadata or not device_metadata.get(b"chassis_serial_number"):
             chassis_serial_number = ""
         else:
             chassis_serial_number = device_metadata[b"chassis_serial_number"]
-            self.physical_classes = [(self.CHASSIS_ID, )]
-            self.physical_classes_map = {
-                    (self.CHASSIS_ID, ): (PhysicalClass.CHASSIS, chassis_serial_number.decode("utf-8"))
-                }
+
+        self.physical_classes_map[chassis_sub_id] = PhysicalClass.CHASSIS
+        self.physical_serial_number_map[chassis_sub_id] = chassis_serial_number
+
+        # retrieve the initial list of transceivers that are present in the system
+        transceiver_info = self.statedb.keys(self.statedb.STATE_DB, self.TRANSCEIVER_KEY_PATTERN)
+        if transceiver_info:
+            self.transceiver_entries = [entry.decode() \
+                for entry in transceiver_info]
+        else:
+            self.transceiver_entries = []
+
+        # update cache with initial data
+        for transceiver_entry in self.transceiver_entries:
+            # extract interface name
+            interface = transceiver_entry.split(mibs.TABLE_NAME_SEPERATOR_VBAR)[-1]
+            self._update_transceiver_cache(interface)
 
     def update_data(self):
         """
         Update cache.
-        NOTE: Nothing to update right now. Implementation is required by framework.
+        Here we listen to changes in STATE_DB TRANSCEIVER_INFO table
+        and update data only when there is a change (SET, DELETE)
+        """
+
+        # This code is not executed in unit test, since mockredis
+        # does not support pubsub
+        if not self.pubsub:
+            redis_client = self.statedb.get_redis_client(self.statedb.STATE_DB)
+            db = self.statedb.db_map[self.statedb.STATE_DB]["db"]
+            self.pubsub = redis_client.pubsub()
+            self.pubsub.psubscribe("__keyspace@{}__:{}".format(db, self.TRANSCEIVER_KEY_PATTERN))
+
+        while True:
+            msg = self.pubsub.get_message()
+            if msg:
```

Comment:
>if msg: [](start = 12, length = 7)

suggest simple refactoring:
exit early if there is no msg.
so the remaining code could be one level less indented.